p2p/http/libp2phttp.go

```diff
@@ -657,12 +660,22 @@ func (h *Host) RoundTrip(r *http.Request) (*http.Response, error) {
 		return nil, err
 	}
 	addr, isHTTP := normalizeHTTPMultiaddr(addr)
```

Comment:
This panics for input `multiaddr:/https/http-path/xyz`

Reply:
Do you also find it surprising that the multiaddr library returns a nil Multiaddr type, and panics when given the type it just returned? I think this variant should hold:

```
before, after := ma.Split(original, ...)
roundTrip := ma.Join(before, after)
roundTrip == original // sometimes true, sometimes panics
```

Reply:
I completely agree. This should always work
```
before, after := ma.SplitFirst(addr)
addr.Equal(ma.Join(before, after))
before, after = ma.SplitLast(addr)
addr.Equal(ma.Join(before, after))
```

We should change `ma.Join` to just ignore nils. I don't think it would break anyone but would help with a lot of such cases.